### PR TITLE
Create test cases for RelatedBookCommand and BookRelatedToPatronPredicate

### LINKED ISSUE #98: Create test cases for RelatedBookCommand and RelatedBookCommandParser
- Add test cases for RelatedBookCommand @yl-ang 
- Add test cases for RelatedBookCommandParser @yuqitanyq

### PR DESCRIPTION
**Pull request by yl-ang**
- Add tests for RelatedBookCommand 
- Add tests for BookRelatedToPatronPredicate
- Add tests for CommandResult to improve code coverage
- Fix minor issues such as renaming variables in BookCardList and test cases

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/CommandResultTest.java [hunk 1]
```diff
@@ -35,6 +35,44 @@ public void equals() {
...
+        CommandResult commandResult = new CommandResult("Here is your previous command.", false,
+                false, true);
+        assertTrue(commandResult.isPrevious());
+    }
+
+    @Test
+    public void test_isPrevious_returnFalse() {
+        CommandResult commandResult = new CommandResult("feedback", false, false, false);
+        assertFalse(commandResult.isPrevious());
+    }
```
Comment (yuqitanyq) [inline src/test/java/seedu/address/logic/commands/CommandResultTest.java:74]: Great job on adding in additional test cases for the command result class. I like how it is done for details such as the help and exit commands. 👍

File: src/test/java/seedu/address/logic/commands/book/RelatedBookCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    public void execute_invalidPatronIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPatronList().size() + 1);
+        RelatedBookCommand relatedBookCommand = new RelatedBookCommand(outOfBoundIndex);
+
+        assertCommandFailure(relatedBookCommand, model, Messages.MESSAGE_INVALID_PATRON_DISPLAYED_INDEX);
+    }
+
+    @Test
```
Comment (yuqitanyq) [inline src/test/java/seedu/address/logic/commands/book/RelatedBookCommandTest.java:44]: Perhaps we could include a test for negative indexes as well so that we can be sure that the command will fail as well for negative indexes? Other than that, I think these tests quite sufficient for the logic of book-related command. 👍

**Review by yuqitanyq — APPROVED** [review]
LGTM 👍 If the additional test case for negative indexes is added in, it would be perfect. Other than that, I think this PR is good to merge! Great job!

File: src/test/java/seedu/address/logic/commands/book/RelatedBookCommandTest.java (hunk 2 above)
Comment (yl-ang) [reply src/test/java/seedu/address/logic/commands/book/RelatedBookCommandTest.java:44]: Thank you for the advice, will add some negative indexes test cases later on 💯

File: src/test/java/seedu/address/logic/commands/CommandResultTest.java [hunk 3]
```diff
@@ -35,6 +35,44 @@ public void equals() {
...
+
```
Comment (aditi2313) [inline src/test/java/seedu/address/logic/commands/CommandResultTest.java:75]: LGTM!

File: src/test/java/seedu/address/model/book/BookRelatedToPatronPredicateTest.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        // Zero patrons
+        BookRelatedToPatronPredicate predicate = new BookRelatedToPatronPredicate(null);
+        assertFalse(predicate.test(BookBuilder.bookBuilderWithBorrowedStatus().build()));
+
+        // Non matching patron
+        Patron predicatePatron = SampleDataUtil.getSamplePatrons()[1];
+        predicate = new BookRelatedToPatronPredicate(predicatePatron);
+        assertFalse(predicate.test(BookBuilder.bookBuilderWithBorrowedStatus().build()));
+    }
+}
```
Comment (aditi2313) [inline src/test/java/seedu/address/model/book/BookRelatedToPatronPredicateTest.java:59]: Looks good!

**Review by aditi2313 — APPROVED** [review]
